### ISSUE TITLE
Driver fail to list datasets under certain circumstances

### ISSUE THREAD

**Issue by angarod** [openebs/zfs-localpv#496]
**What steps did you take and what happened:**
Create a persistent volume claim and a persistent volume on a server that use zfs as filesystem.
ZFS topology:
```
$ zfs list
nvme                                242K  9.20G       24K  none
nvme/openebs                        120K  9.20G       24K  none
nvme/openebs/dataset                24K   9.20G       24K  none
```

**What did you expect to happen:**
PVC and PV are successfully created without error.

**Anything else you would like to add:**
The driver cannot list datasets since depth level is hard coded to 1, which prevents the driver from reading my topology. Seems to be related to https://github.com/openebs/zfs-localpv/commit/ba0e1749ec1841bb8f9d704225e6df04787dc608.
`openebs-zfs-plugin E1229 09:29:32.162493       1 zfs_util.go:931] zfs: could not list zpools cmd [list -d 1 -s name -o name,guid,available -H -p]: exit status 127  `

**Environment:**
- ZFS-LocalPV version: `zfs-localpv-2.3.0`
- Kubernetes version (use `kubectl version`): `v1.28.2`
- Kubernetes installer & version: `v1.28.2`
- Cloud provider or hardware configuration: bare-metal
- OS (e.g. from `/etc/os-release`): `Ubuntu 20.04.6 LTS`

**Comment by Val** [comment]
@lowjoel what do you propose to meet the need for topology on several levels?

**Comment by avishnu** [comment]
Yes, this could be the issue. Any takers for fixing? Possible approaches could be determine the depth based on the input dataset or checking for the existence of the input dataset..

**Comment by avishnu** [comment]
There has been no further activity on this issue. Moving out-of-scope from OpenEBS v4.3.

**Comment by krishnaGajabi** [comment]
@angarod / @Abhinandan-Purkait , I am thinking to take a look at this issue.

**Comment by Abhinandan-Purkait** [comment]
@krishnaGajabi Thanks! Please feel free to assign yourself.
